sammo.py

```diff
@@ -149,92 +144,71 @@ def onCreateSession(self, sessionDirectory: str) -> None:
 
         self.soundRecordingController.onNewSession(sessionDirectory)
 
+        self.tableDock.init(
+            self.session.environmentLayer, self.session.sightingsLayer
+        )
+        self.tableDock.refresh(self.session.sightingsLayer)
+        self.tableDock.refresh(self.session.environmentLayer)
+
         # init simu
         if self.simuGpsAction:
             self.simuGpsAction.onNewSession()
 
     def onEffortAction(self, onEffort: bool):
-        if not onEffort:
-            if self.updateEffort("E"):  # cancel
-                self.statusDock.isEffortOn = False
-            else:
-                self.effortAction.action.setChecked(True)
-        else:
-            if not self.updateEffort("B"):
-                # the user pressed the CANCEL button of the form
-                self.effortAction.action.setChecked(False)
-                self.statusDock.isEffortOn = False
-            else:
-                self.statusDock.isEffortOn = True
-
-        self.environmentAction.onChangeEffortStatus(self.statusDock.isEffortOn)
-
-    def updateEffort(self, status: str = "A") -> bool:
         self.soundRecordingController.onStartEnvironment()
-
         layer = self.session.environmentLayer
         feat = QgsVectorLayerUtils.createFeature(layer)
-        layer.startEditing()
-        layer.addFeature(feat)
         feat["dateTime"] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
-        feat["status"] = status
 
-        layer.startEditing()
-        if self.iface.openFeatureForm(layer, feat):
-            layer.updateFeature(feat)
-            layer.commitChanges()
-            self.soundRecordingController.onStopEventWhichNeedSoundRecord()
-            return True
-        else:
-            self.soundRecordingController.hardStopOfRecording()
-            layer.rollBack()
-            return False
+        if not layer.isEditable():
+            layer.startEditing()
+        layer.addFeature(feat)
+
+        self.saveAll()
+
+        self.tableDock.refresh(layer)
+        self.soundRecordingController.onStopEventWhichNeedSoundRecord()
 
     def onObservationAction(self):
         self.soundRecordingController.onStartObservation()
 
         layer = self.session.sightingsLayer
         feat = QgsVectorLayerUtils.createFeature(layer)
-        layer.startEditing()
-        layer.addFeature(feat)
-        if self.session.lastGpsGeom:
-            feat.setGeometry(self.session.lastGpsGeom)
```

Comment:
@pblottiere I didn't pay attention on this the other day because it was not the purpose of the PR, but here the PR erased the geometry setter function on Observation feature (and it's happened the same on Follower). Since then, our Observer entities have no geometry and don't appear on canvas.